src/main/java/com/iexec/core/task/TaskExecutorEngine.java

```diff
@@ -24,12 +28,21 @@ public TaskExecutorEngine(TaskService taskService) {
         executorMap = new ConcurrentHashMap<>();
     }
 
-    public void updateTask(String chainTaskId) {
+    public CompletableFuture<Boolean> updateTask(String chainTaskId) {
 
         executorMap.putIfAbsent(chainTaskId, ThreadPoolExecutorUtils.singleThreadExecutorWithFixedSizeQueue(1));
 
         Executor executor = executorMap.get(chainTaskId);
-        executor.execute(() -> taskService.tryUpgradeTaskStatus(chainTaskId));
+
+        return CompletableFuture.supplyAsync(() -> taskService.tryUpgradeTaskStatus(chainTaskId), executor)
```

Comment:
Why do we need to return a completableFuture here? It seems that returning nothing is good enough no?

Reply:
because in some cases I needed to wait for the execution to end.
Exp: ReplicateSupplyService > recoverReplicateInRevealPhase > L259